sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/SetOperationSuite.scala

```diff
@@ -76,4 +77,32 @@ class SetOperationSuite extends PlanTest {
         testRelation3.select('g) :: Nil).analyze
     comparePlans(unionOptimized, unionCorrectAnswer)
   }
+
+  test("no more unnecessary distincts in multiple unions") {
+    val query1 = OneRowRelation
+      .select(Literal(1).as('a))
+    val query2 = OneRowRelation
+      .select(Literal(2).as('b))
+    val query3 = OneRowRelation
+      .select(Literal(3).as('c))
+
+    val unionQuery1 = Distinct(Union(Distinct(Union(query1, query2)), query3)).analyze
```

Comment:
```
D - U - D - U - query1
    |       |
   query3  query2
```
Could you add a comment about the tree structure of your test case? It can help the reviewers understand your test cases. Thanks!